src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -68,20 +69,60 @@ public void addAnswer(Answer answer) {
         answers.add(answer);
     }
 
-    public boolean isOwner(NsUser loginUser) {
+    //region [checkDeletePermission]
+    public void checkDeletePermission(NsUser loginUser) throws CannotDeleteException {
+        if (isNotOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+    }
+
+    private boolean isOwner(NsUser loginUser) {
         return writer.equals(loginUser);
     }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
+    private boolean isNotOwner(NsUser loginUser){
+        return !isOwner(loginUser);
+    }
+    //endregion
+
+    public List<DeleteHistory> delete(NsUser loginUser) throws CannotDeleteException {
+        validate(loginUser);
+        markDeleted();
+        return getHistory();
+    }
+
+    public void markDeleted() {
+        this.deleted = true;
+    }
+
+    private void validate(NsUser loginUser) throws CannotDeleteException {
+        checkDeletePermission(loginUser);
+        checkAnswerDeletePermission(loginUser);
+    }
+
+    private void checkAnswerDeletePermission(NsUser loginUser) throws CannotDeleteException {
+        Answers answers = getAnswers();
+        answers.checkDeletePermission(loginUser);
+    }
+
+    private List<DeleteHistory> getHistory() {
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
+        deleteHistories.add(QuestionDeleteHistory());
+
+        List<DeleteHistory> answerHistories = answers.delete();
+        deleteHistories.addAll(answerHistories);
+        return deleteHistories;
+    }
+
+    private DeleteHistory QuestionDeleteHistory() {
```

Comment:
메서드 이름은 소문자로 시작하도록 작성해보면 어떨까요?
```suggestion
    private DeleteHistory questionDeleteHistory() {
```

Reply:
기본적인건데 지키지 못했네요;; 수정했습니다